File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start)
{
    int c = 0;
    int n = start;

    if(n < 1)
    {
        return -1;
    }

    while (n > 1)
    {
        switch (n % 2)
        {   
        case 0: 
            n = n/2;
            c++;
            break;
        default:
            n = 3*n+1;
            c++;
            break;
        }
    }
    return c;
}

File: csharp/collatz-conjecture/CollatzConjecture.cs
public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number < 1)
        {
            throw new ArgumentOutOfRangeException();
        }
        int n = 0;
        while (number > 1)
        {
            if (number % 2 == 0)
            {
                number = number / 2;
                n++;
            }
            else
            {
                number = (number * 3) + 1;
                n++;
            }
        }
        return n;
    }
}
